frontend/src/App.tsx

```diff
@@ -1,7 +1,78 @@
-import './App.css';
+import clsx from 'clsx';
+import style from './App.module.css';
+import {
+  Carousel,
+  CarouselIndicatorList,
+  CarouselSlide,
+} from './components/Carousel';
+import { slides } from './fixtures';
 
 function App() {
-  return <div className='App'>{/* write your component here */}</div>;
+  return (
+    <Carousel
+      interval={2000}
+      className='relative'
+      aria-label='yangbot-homework'
+    >
+      {(ctx) => {
```

Comment:
Use render props here is to access `currentIndex` to build its own animation effects
After some thinkings, render props can be replaced with different approaches here

1. emit onChange in `<Carousel />` and let user set its own state for customization
2. provide another compound component `<Carousel.Content getStyles={ctx => ({}) />`

Personally,  I lean towards minimizing the number of compound components.... either render props or maintain own state would suffice in this context and provide more flexibility. Correct me if I am wrong :P

Demo's Carousel children api can be improved as

`{ children: ReactNode | ((ctx: CarouselContext) => ReactNode) }`